File: tested/languages/haskell/templates/Selector.hs
module Selector where

import System.Environment

## In Haskell, we import all modules up front, since there is no
## dynamic module loading (or at least not an easy way).
% for c in contexts:
    import qualified ${c}
% endfor

main = do
    [n] <- getArgs
    case n of
        % for c in contexts:
            "${c}" -> ${c}.main
        % endfor

File: tested/languages/csharp/templates/Selector.cs
<%page args="contexts" />
using System;

namespace Tested {
    class Selector {
        public static void Main(string[] args) {
            String name = args[0];
            % for c in contexts:
                if ("${c}" == name) {
                    ${c}.Main(Array.Empty<String>());
                }
            % endfor
        }
    }
}
